tensorboard/webapp/runs/store/runs_selectors.ts

```diff
@@ -72,27 +74,48 @@ export const getRun = createSelector(
  */
 export const getRuns = createSelector(
   getDataState,
-  (state: RunsDataState, props: {experimentId: string}): Run[] => {
+  getRunsToHparamsAndMetrics,
+  (
+    state: RunsDataState,
+    runsToHparamsAndMetrics: RunToHparamsAndMetrics,
+    props: {experimentId: string}
+  ): Run[] => {
     const runIds = state.runIds[props.experimentId] || [];
     return runIds
       .filter((id) => Boolean(state.runMetadata[id]))
-      .map((id) => state.runMetadata[id]);
+      .map((id) => ({...state.runMetadata[id]}))
+      .map((metadata) => {
+        if (runsToHparamsAndMetrics[metadata.id]) {
```

Comment:
I had hoped the changes in the data model would be reflected in the other layers as well - in how we model our selectors and how the components retrieve data. I had hoped that dashboard things would clearly rely on dashboard hparams data (HparamsState.dashboardSpec and dashboardSessionGroups) while experiment list things would rely on experiment list data (RunsState.runMetadata.hparams and metrics).

Instead we've quietly merged them in this layer, making it difficult to predict whether a runs table in the experiment list, for example, will be getting its data from one part of the data model or another.

More concretely, could we refactor hparams value retrieval into two selectors: 
* getExperimentHparamsAndMetrics, which accepts an experimentId as property and returns hparam and metric values from the old location (ie RunsState.runMetadata)
* getDashboardHparamsAndMetrics, which requires no properties and returns hparam and metric values from the new location (ie HparamsState.dashboardSessionGroups).

The first selector would be called for runs tables in the experiment list, the second selector would be called for runs tables in the dashboard.

Similarly (but differently) could the runs table in the dashboard rely more heavily on the data we now store in HparamsState.dashboardSpecs? Is it correct to say that this current scheme will still rely on us (re)constructing the domain information using the run information retrieved here? Would it be clearer to instead take the data already returned to you by the server and use that as the domain information?

Reply:
We spoke offline. The key insights:

* getRuns is essentially getExperimentHpramsAndMetrics. It is only used by the runs tables in the experiment list.
* getRunsFromExperimentIds is essentially getDashboardHparamsAndMetrics. It is only used by the new runs table and scalar card data tables in the dashboard.
* we will be using HparamsState.dashboardSpecs to generate hparam filter domains. We don't see that happening in this PR because we don't have filters in the new runs table, yet.

Action items:
* We agreed these functions need to be renamed. If we decide to rename in a later PR, it would at least be nice to put a comment in this PR.
* We agreed that getRuns should only use hparams and metrics data from RunsState.
* We agreed that getRunsFromExperimentIds should only use hparams and metrics data from HparamsState.